stages/org.osbuild.breadcrumb.image-builder

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python3
+"""
+The `org.osbuild.breadcrumb.image-builder` stage is a breadcrumb stage to put
+artifacts in the built image that can identify the `image-builder` build-job
+and settings used to produce the image as built by `osbuild`.
+"""
+
+import os
+import sys
+import json
+
+import osbuild.api
+
+
+SCHEMA = r"""
+"definitions": {
+  "uuid": {
+    "type": "string",
+    "pattern": "^[a-zA-Z0-9-]+$"
+  }
+},
+"properties": {
+  "build-identifier": {
+    "$ref": "#/definitions/uuid",
+    "description": "A unique identifier to look up the related build."
+  }
+}
+"""
+
+
+def main(tree, options):
+    path = os.path.join(tree, "usr", "share", "osbuild", "self")
+    file = os.path.join(path, "image-builder.json")
```

Comment:
I'm undecided on this as well, both on names of the file(s), location of the file(s), and if it should be a single file or separate ones.

Including version information is something I do want to do (though I don't know if all tools can interrogate their own version number).

Reply:
I feel there are three potential kinds of things to put into the image:

- the manifest (how was the image built) - out of scope for this PR?
- version information about the build tools (in what environment was it built) - also out of scope?
  - the osbuild git sha
  - the host kernel version?
  - the image definition git sha (i.e., composer as long as that's where the image info lives)
- information about what/how the user requested the image - I think this is the most relevant right now
  - is the origin of the image `on-prem`, `marketplace` or `service` (probably not the right labels)?
  - what sort of configuration was applied at a high level? (modified filesystem, non-RHEL content, etc). think about the kind of questions we might want to ask for high-level business analysis (for instance to determine whether certain features are used / useful)

I think a simple JSON file with key/value pairs might give us all we need, but something more clever might be worth it (I suggest focusing on the third point for now though).

Reply:
> the manifest (how was the image built) - out of scope for this PR?

I think that could be a different stage even, and we write the manifest into `self/manifest.json`

> I think a simple JSON file with key/value pairs might give us all we need

Yeah, the question here was more like, how should that file be called and how tight should we pin down the schema for the key-value pairs. I mean we can always have it free form, but I feel that defining a proper schema for it would be better since client tooling will require the key, values to be in an a well defined form and if that exists we might as well put it into the schema here to so clients can rely that it does not change. 

A hybrid approach would be to have some well known information `version` and such well defined in the file and then allow for an `image-builder` section that is free form and we just pass on as-is.

Reply:
See https://github.com/osbuild/osbuild/pull/1060#issuecomment-1190449770